azkaban-common/src/main/java/azkaban/user/UserUtils.java

```diff
@@ -22,4 +37,67 @@ public static boolean hasPermissionforAction(final UserManager userManager, fina
 
     return false;
   }
+
+  /**
+   * Creates a watch thread which listens to specified files' modification and reloads
+   * configurations
+   */
+  static void setupWatch(final Map<String, ParseConfigFile> configFileMap) {
+    Preconditions.checkArgument(configFileMap != null);
+
+    Runnable runnable = () -> {
+      WatchService watchService;
+      Path path;
+      for (Map.Entry<String, ParseConfigFile> entry : configFileMap.entrySet()) {
+        String fileName = entry.getKey();
+        ParseConfigFile parser = entry.getValue();
+        Preconditions.checkArgument(fileName != null && parser != null);
+
+        final File file = new File(fileName);
+        if (!file.exists()) {
+          throw new IllegalArgumentException("User provided file " + fileName + " does not exist.");
+        }
+
+        try {
+          watchService = FileSystems.getDefault().newWatchService();
+          path = Paths.get(fileName).getParent();
+          path.register(watchService, new WatchEvent.Kind[]{StandardWatchEventKinds.ENTRY_MODIFY},
+              SensitivityWatchEventModifier.HIGH);
+        } catch (IOException e) {
+          // Ignore the IOException
+          log.warn("IOException while setting up watch on conf"
+              + e.getMessage());
+          return;
+        }
+        for (; ; ) {
+          // Watch for modifications
+          WatchKey watchKey;
+          try {
+            watchKey = watchService.take();
+          } catch (InterruptedException ie) {
+            log.warn(ie.getMessage());
```

Comment:
This link should work: https://stackoverflow.com/questions/3976344/handling-interruptedexception-in-java

Please call Thread.currentThread().interrupt(); to set the interrupt flag.